pkg/controller/operconfig/ingressconfig_controller.go

```diff
@@ -0,0 +1,90 @@
+type ReconcileIngressConfigs struct {
+	client client.Client
+}
+
+// Reconcile sets the openshift-host-network namespaces' labels as per the
+// endpointPublishingStrategy of the `default` ingress controller object.
+func (r *ReconcileIngressConfigs) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	if request.Namespace == names.IngressControllerNamespace && request.Name == names.DefaultIngressControllerName {
+		log.Printf("Reconciling update to IngressController %s/%s\n", request.Namespace, request.Name)
+		ingressControllerConfig := &operv1.IngressController{TypeMeta: metav1.TypeMeta{APIVersion: operv1.GroupVersion.String(), Kind: "IngressController"}}
+		err := r.client.Get(context.TODO(), request.NamespacedName, ingressControllerConfig)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Printf("Ingress Controller configuration %s was deleted", request.NamespacedName.String())
+				// Request object not found, could have been deleted after reconcile request.
+				// Owned objects are automatically garbage collected, since we set
+				// the ownerReference (see https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/).
+				// Return and don't requeue
+				return reconcile.Result{}, nil
+			}
+			// Error reading the object - requeue the request.
+			log.Printf("Unable to retrieve IngressController.operator.openshift.io object: %v", err)
+			return reconcile.Result{}, err
+		}
+		addLabel := ingressControllerConfig.Spec.EndpointPublishingStrategy != nil &&
+			ingressControllerConfig.Spec.EndpointPublishingStrategy.Type == operv1.HostNetworkStrategyType
+
+		err = r.updatePolicyGroupLabelOnNamespace(names.HostNetworkNamespace, addLabel)
+		if err != nil {
+			log.Printf("Error setting the host network label on namespace %s: %v", names.HostNetworkNamespace, err)
+			return reconcile.Result{}, err
+		}
+		return reconcile.Result{RequeueAfter: ResyncPeriod}, nil
+	}
+	return reconcile.Result{}, nil
+}
+
+// setLabelsOnNamespace sets the labels specified on the target namespace using the client API
+func (r *ReconcileIngressConfigs) updatePolicyGroupLabelOnNamespace(targetNamespace string, add bool) error {
+	var err error
+	namespace := &corev1.Namespace{TypeMeta: metav1.TypeMeta{APIVersion: corev1.SchemeGroupVersion.String(), Kind: "Namespace"}}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: targetNamespace}, namespace)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		// Error reading the object - requeue the request.
+		log.Printf("Unable to retrieve namespace object: %v", err)
+		return err
+	}
+	newNamespace := namespace.DeepCopy()
+	existingLabels := newNamespace.GetLabels()
+	if existingLabels == nil {
+		existingLabels = map[string]string{}
+	}
+	if !add {
+		delete(existingLabels, names.PolicyGroupLabelIngress)
+	} else {
+		existingLabels[names.PolicyGroupLabelIngress] = names.PolicyGroupLabelIngressValue
```

Comment:
we should probably set/delete both the legacy and new label names/values

Reply:
we need additional context then if this is for `openshift-sdn` or `ovn-k`? I don't see a reason to touch existing labels here. In upgrade scenarios, we should preserve the existing labels if any on the default namespace and this doesn't touch that namespace at all.

Reply:
no, I mean, we should set both `network.openshift.io/policy-group: ingress` and `policy-group.network.openshift.io/ingress: ""` on the namespace, so that it works both with new NetworkPolicies targeting the new label name and with existing NetworkPolicies targeting the old label name.